apps/06.1Checkers.js

```diff
@@ -51,16 +56,73 @@ function Board() {
     }
     console.log(string);
   };
-
-  // Your code here
+  //code here ------------
+  this.createCheckers = function() {
+    var whiteCheckers = [
+      [0, 1], [0, 3], [0, 5], [0, 7],
+      [1, 0], [1, 2], [1, 4], [1, 6],
+      [2, 1], [2, 3], [2, 5], [2, 7]
+    ];
+    var blackCheckers = [
+      [5, 0], [5, 2], [5, 4], [5, 6],
+      [6, 1], [6, 3], [6, 5], [6, 7],
+      [7, 0], [7, 2], [7, 4], [7, 6]
+    ];
+    //create an array of white checkers and their positions;
+    for (var i = 0; i < whiteCheckers.length; i++) { // for (var position of whiteCheckers)
+      var position = whiteCheckers[i];
+      var whiteChecker = new Checker ('white');
+      //assign a whiteChecker to grid at position[0,1]
+      this.grid[position[0]][position[1]] = whiteChecker;
+      this.checkers.push(whiteChecker);
+    }
+    //create an array of black checkers and their positions
+    for (var i = 0; i < blackCheckers.length; i++) {
+      var position = blackCheckers[i];
+      var blackChecker = new Checker ('black');
+      //assign a whiteChecker to grid at position[0,1]
+      this.grid[position[0]][position[1]] = blackChecker;
+      this.checkers.push(blackChecker);
+    }
+  };
+  this.selectChecker = function(position) {
+    return this.grid[position[0]][position[1]];
+  };
+  this.killChecker = function(position) {
+    var checkerToKill = this.selectChecker(position);
+    var indexOfChecker = this.checkers.indexOf(checkerToKill);
+    if (checkerToKill !== null) {
+      this.checkers.splice(indexOfChecker, 1);
+      //find the index of 'position' in this.checkers[];
+      this.grid[position[0]][position[1]] = null;
+      //assign this position to null; in this.grid -> this.grid(position) = null;
+    }
+  }
 }
 function Game() {
-
   this.board = new Board();
-
   this.start = function() {
     this.board.createGrid();
-    // Your code here
+    this.board.createCheckers();
+  };
+  this.moveChecker = function(start, end) {
+    start = start.split("");
```

Comment:
JavaScript treats simple strings as an array of characters so there's not need to use the split() method to scrub them.